src/pyhf/modifiers/normfactor.py

```diff
@@ -74,6 +73,8 @@ def apply(self, pars):
             )
 
         results_normfactor = tensorlib.where(
-            self.normfactor_mask, results_normfactor, self.normfactor_default
+            tensorlib.astensor(self.normfactor_mask, dtype="bool"),
```

Comment:
same here.. could just wrap the `tensorlib.tile` call in precompute to do the cast